src/program/snabblwaftr/run/run.lua

```diff
@@ -0,0 +1,149 @@
+       --v4-pci <pci-addr>      PCI device number for the INET-side NIC
+       --v6-pci <pci-addr>      PCI device number for the B4-side NIC
+       -D       <seconds>       Duration in seconds
+       -v                       Verbose
+       -u                       Ultraverbose
+]])
+end
+
+local function fatal(msg)
+   usage()
+   print(msg)
+   main.exit(1)
+end
+
+local function print_info(app, link)
+   if ultra_verbose then
+      app.report_apps()
+   end
+end
+
+local long_opts = {
+   ["bt"]           = "b",
+   ["conf"]         = "c",
+   ["duration"]     = "D",
+   ["help"]         = "h",
+   ["ultraverbose"] = "u",
+   ["v4-pci"]       = "n",
+   ["v6-pci"]       = "m",
+   ["verbose"]      = "v",
+}
+
+-- Options
+local bt_file, conf_file, v4_pci, v6_pci
+local duration, ultra_verbose
+
+-- Verbose by default
+local verbose = true
+
+local function nic_ui (args)
+   local bt_file, conf_file, inet_nic_pci, b4side_nic_pci = unpack(args)
+
+   -- It's essential to initialize the binding table before the aftrconf
+   bt.get_binding_table(bt_file)
+   local aftrconf = conf.get_aftrconf(conf_file)
+
+   local c = config.new()
+   config.app(c, 'inetNic', Intel82599, {pciaddr=inet_nic_pci,
+                                         macaddr = ethernet:ntop(aftrconf.aftr_mac_inet_side)})
+   config.app(c, 'b4sideNic', Intel82599, {pciaddr=b4side_nic_pci,
+                                           macaddr = ethernet:ntop(aftrconf.aftr_mac_b4_side)})
+   config.app(c, 'lwaftr', lwaftr.LwAftr, aftrconf)
+   config.app(c, 'v6_stats', basic_apps.Statistics)
+   config.app(c, 'v4_stats', basic_apps.Statistics)
+
+   config.link(c, 'inetNic.tx -> lwaftr.v4')
+   config.link(c, 'b4sideNic.tx -> lwaftr.v6')
+   if verbose then
+      config.link(c, 'lwaftr.v4 -> v4_stats.input')
+      config.link(c, 'v4_stats.output -> inetNic.rx')
+      config.link(c, 'lwaftr.v6 -> v6_stats.input')
+      config.link(c, 'v6_stats.output -> b4sideNic.rx')
+   else
+      config.link(c, 'lwaftr.v4 -> inetNic.rx')
+      config.link(c, 'lwaftr.v6 -> b4sideNic.rx')
+   end
+
+   engine.configure(c)
+
+   if ultra_verbose then
+      local function lnicui_info() print_info(engine) end
+      local t = timer.new("report", lnicui_info, 1e9, 'repeating')
+      timer.activate(t)
+   end
+
+   if duration then
+      engine.main({duration=duration, report={showlinks=true}})
+   else
+      engine.main({report={showlinks=true}})
+   end
+end
+
+function run (parameters)
+   if #parameters == 0 then
+      usage()
+      main.exit(1)
+   end
+   local opt = {}
+   function opt.v (arg) verbose = true  end
+   function opt.u (arg) ultra_verbose = true verbose = true end
+   function opt.D (arg) duration = tonumber(arg)  end
+   function opt.b (arg) bt_file = arg end
+   function opt.c (arg) conf_file = arg end
+   function opt.n (arg) v4_pci = arg end
+   function opt.m (arg) v6_pci = arg end
+   function opt.h () usage() main.exit(1) end
```

Comment:
Small nit: The convention for Unix tools when `--help` or `-h` is passed as argument is to print the usage summary text, then exit with a `0` status. The rationale is that the execution is successful because the user asked for help, and the tool provided help successfully, so the exist status should be zero to signal success. For example, not how `ls --help` exits with a “success” status code:
```
% ls --help &> /dev/null ; echo $?
0
% ls --this-does-not-exist &> /dev/null ; echo $?
2
%
```